From f365f781dabe87af8e3cc09794f4977cc7838ea0 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep Bhatta <subbaraya.sundeep.bhatta@xilinx.com>
Date: Thu, 14 Mar 2013 16:19:34 +0530
Subject: [PATCH 519/628] usb: gadget: Added required header files

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit ba3811b89f2d9040c5d2945811b9ebe3405e21f3

Added required header files.
Replaced : SA_RESTART -> IRQF_SHARED

Signed-off-by: Subbaraya Sundeep Bhatta <sbhatta@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/gadget/xilinx_udc.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/xilinx_udc.c b/drivers/usb/gadget/xilinx_udc.c
index cdaffb8..6e9b891 100644
--- a/drivers/usb/gadget/xilinx_udc.c
+++ b/drivers/usb/gadget/xilinx_udc.c
@@ -25,13 +25,16 @@
 
 #include <linux/interrupt.h>
 #include <linux/device.h>
+#include <linux/module.h>
+#include <linux/prefetch.h>
 #include <linux/usb/ch9.h>
 #include <linux/usb/gadget.h>
 #include <linux/io.h>
-#include <linux/irq.h>
 #include <linux/seq_file.h>
+#include <linux/of_address.h>
 #include <linux/of_device.h>
 #include <linux/of_platform.h>
+#include <linux/of_irq.h>
 #include <linux/dma-mapping.h>
 #include "gadget_chips.h"
 
@@ -2285,7 +2288,7 @@ static int xudc_init(struct device *dev, struct resource *regs_res,
 
 	/* Request UDC irqs */
 	if (request_irq
-	    (irq_res->start, xusb_udc_irq, SA_RESTART, driver_name, udc)) {
+	    (irq_res->start, xusb_udc_irq, IRQF_SHARED, driver_name, udc)) {
 		device_unregister(&udc->gadget.dev);
 		stop_activity(udc);
 		iounmap(udc->base_address);
-- 
1.7.5.4

